File: resolucoes/kotlin/lista01/src/main/kotlin/ex002.kt
import java.util.*

/**
 * Faça um programa que leia o nome de uma pessoa e mostre uma mensagem de boasvindas para ela:
 * Ex:
 * Qual é o seu nome? João da Silva
 * Olá João da Silva, é um prazer te conhecer!
 * */
fun ex002(){
    val scanner = Scanner(System.`in`)
    print("Qual é o seu nome? ")
    val nome = scanner.nextLine()
    println("Olá $nome é um prazer te conhecer!")
}

internal fun main(){
    ex002()
}

File: resolucoes/C++/lista01/ex002.cpp
#include <iostream>

int main() {
    // Definição da variavel do tipo string
    // os tipos das variaveis do C++ precisam ser explicitamente declarados
    std::string nome_da_pessoa;

    // printando ... para cout (output padrão)
    std::cout << "Qual é o seu nome? ";

    // lendo o valor de cin (input padrão) para variavel ...
    std::cin >> nome_da_pessoa;

    // printando resultado, note que os texto e a variavel são separados por <<
    // no objeto cout isso significa que estamos escrevendo para o output padrão
    // Já no >> em cin significa que estamos lendo do input padrão
    std::cout << "Olá " << nome_da_pessoa << ", é um prazer te conhecer!\n";
    
    return 0;
}
